### PR TITLE
Move QnAMaker Dialog/Recognizer out of Adaptive and into QnA assembly

### PR DESCRIPTION
**Pull request by tomlm**
#3719 

* Merge QnAMakerDialog2 => QnAMakerDialog 
* Move QnAMakerDialog and QnAMakerRecognizer into QnA assembly
* Move schema definitions int QnA assembly
* Move ComponentRegistration into QnA Assembly
* remove dependencies between Adaptive and QnA
* NOTE: Needed to move ITemplate<> into a shared location as that was only dependency on Adaptive, so moved it into BotBuilder which is where IRecognizer lives.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog2.cs [hunk 1]
```diff
@@ -0,0 +1,258 @@
+﻿//using System.Collections.Generic;
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog2.cs:1]: I don't think this is right.

File: libraries/Microsoft.Bot.Builder.AI.QnA/Microsoft.Bot.Builder.AI.QnA.csproj [hunk 2]
```diff
@@ -28,22 +28,26 @@
...
     <PackageReference Include="Microsoft.Bot.Builder.Dialogs" Condition=" '$(IsBuildServer)' != '' " Version="$(ReleasePackageVersion)" />
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All"/>
+    <PackageReference Include="Microsoft.Bot.Builder.Dialogs.Declarative" Condition=" '$(IsBuildServer)' == '' " Version="4.8.0-local" />
+    <PackageReference Include="Microsoft.Bot.Builder.Dialogs.Declarative" Condition=" '$(IsBuildServer)' != '' " Version="$(ReleasePackageVersion)" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
   </ItemGroup>
 
   <ItemGroup>
+		<ProjectReference Include="..\AdaptiveExpressions\AdaptiveExpressions.csproj" />
+		<ProjectReference Include="..\Microsoft.Bot.Builder.Dialogs.Declarative\Microsoft.Bot.Builder.Dialogs.Declarative.csproj" />
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.QnA/Microsoft.Bot.Builder.AI.QnA.csproj:50]: This is a new dependency in this tree. Is this correct?

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerComponentRegistration.cs [hunk 3]
```diff
@@ -27,13 +25,13 @@ public class QnAMakerComponentRegistration : ComponentRegistration, IComponentDe
...
             // Dialogs
-            yield return new DeclarativeType<QnAMakerDialog2>(QnAMakerDialog2.Kind);
+            yield return new DeclarativeType<QnAMakerDialog>(QnAMakerDialog.Kind);
 
             // Recognizers
             yield return new DeclarativeType<QnAMakerRecognizer>(QnAMakerRecognizer.Kind);
 
             // Triggers
-            yield return new DeclarativeType<OnQnAMatch>(OnQnAMatch.Kind);
+        //    yield return new DeclarativeType<OnQnAMatch>(OnQnAMatch.Kind);
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerComponentRegistration.cs:34]: Do we need this comment.

File: libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog2.cs (hunk 1 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog2.cs:1]: You should remove the file...

---
In reply to: [412565861](https://github.com/microsoft/botbuilder-dotnet/pull/3775#discussion_r412565861) [](ancestors = 412565861)

**Pull request merged**